var/spack/repos/builtin/packages/py-cryptography/package.py

```diff
@@ -14,6 +14,7 @@ class PyCryptography(PythonPackage):
     homepage = "https://pypi.python.org/pypi/cryptography"
     url      = "https://pypi.io/packages/source/c/cryptography/cryptography-1.8.1.tar.gz"
 
+    version('2.7',   sha256='e6347742ac8f35ded4a46ff835c60e68c22a536a8ae5c4422966d06946b6d4c6')
     version('2.3.1', sha256='8d10113ca826a4c29d5b85b2c4e045ffa8bad74fb525ee0eceb1d38d4c70dfd6')
     version('1.8.1', sha256='323524312bb467565ebca7e50c8ae5e9674e544951d28a2904a50012a8828190')
 
```

Comment:
It looks like a lot of these dependency versions have changed:
```python
if (                                                                            
    pkg_resources.parse_version(setuptools.__version__) <                       
    pkg_resources.parse_version("18.5")                                         
):                                                                              
    raise RuntimeError(                                                         
        "cryptography requires setuptools 18.5 or newer, please upgrade to a "  
        "newer version of setuptools"        

setup_requirements = ["cffi>=1.8,!=1.11.3"]                                     

    python_requires='>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*',                    
                                                                                
    install_requires=[                                                          
        "asn1crypto >= 0.21.0",                                                 
        "six >= 1.4.1",                                                         
    ] + setup_requirements,                                                     
    extras_require={                                                            
        ":python_version < '3'": ["enum34", "ipaddress"], 

        "test": [                                                               
            "pytest>=3.6.0,!=3.9.0,!=3.9.1,!=3.9.2",                            
            "pretend",                                                          
            "iso8601",                                                          
            "pytz",                                                             
            "hypothesis>=1.11.4,!=3.79.2",                                      
        ],                                                                      
        "docs": [                                                               
            "sphinx >= 1.6.5,!=1.8.0",                                          
            "sphinx_rtd_theme",                                                 
        ],                                                                      
        "docstest": [                                                           
            "doc8",                                                             
            "pyenchant >= 1.6.11",                                              
            "twine >= 1.12.0",                                                  
            "sphinxcontrib-spelling >= 4.0.1",                                  
        ],                                                                      
        "pep8test": [                                                           
            "flake8",                                                           
            "flake8-import-order",                                              
            "pep8-naming",                                                      
        ],                                                                      
        # This extra is for the U-label support that was deprecated in          
        # cryptography 2.1. If you need this deprecated path install with       
        # pip install cryptography[idna]                                        
        "idna": [                                                               
            "idna >= 2.1",                                                      
        ]                                                                       
    },                               
```
You don't need to add all the extras, but things like idna should be controlled by variants that are off by default.